### PR TITLE
Implement bfa.feast_as_dps, copied from the legion version

### PR DESCRIPTION
**Pull request by OrcinusDrake**
Use legion.feast_as_dps, and copy paste with name changes to bfa. Affects bountiful_captains_feast

### CODE REVIEW

File: engine/player/sc_unique_gear_x7.cpp [hunk 1]
```diff
@@ -74,11 +74,28 @@ void consumables::galley_banquet( special_effect_t& effect )
...
+      effect.trigger_spell_id = 259454;
+      break;
+    case STAT_INTELLECT:
+      effect.trigger_spell_id = 259455;
+      break;
+    default:
+      break;
+  }
+
+  if (effect.player -> role == ROLE_TANK && !effect.player->sim->bfa_opts.bountiful_feast_as_dps )
```
Comment (nuoHep) [inline engine/player/sc_unique_gear_x7.cpp]: You could've simply added this check to `init_feast()`, so it works for both of them.

Comment (nuoHep) [inline engine/player/sc_unique_gear_x7.cpp:92]: Why not do a generic `feast_as_dps` and stuff the check inside `init_feast()`?

Comment (nuoHep) [reply engine/player/sc_unique_gear_x7.cpp:92]: Especially considering the options is already named this way...

Comment (nuoHep) [reply engine/player/sc_unique_gear_x7.cpp:92]: I'm also not sure there is any point in it being expansion-scoped.

**Comment by OrcinusDrake** [comment]
new version here: https://github.com/simulationcraft/simc/pull/4417 cause I don't understand how updating works

**Pull request closed without merging**